### PR TITLE
fix: Move billing limit input to kea-forms

### PR DESCRIPTION
**Pull request by xrdt**
## Problem

List of things that were fixed: 
* input always defaults to 500 when you click to edit it
* can't actually delete all the digits and enter a desired limit value
* users could submit decimal values and screw up the back end 

Other annoying things: 
* Layout shift between editing and non editing states

## Changes

https://github.com/PostHog/posthog/assets/21014901/7328f299-a181-446c-9cfd-dc4c74b6d1bd

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/scenes/billing/billingProductLogic.ts [hunk 1]
```diff
@@ -202,14 +202,17 @@ export const billingProductLogic = kea<billingProductLogicType>([
         ],
     })),
     listeners(({ actions, values, props }) => ({
-        loadBillingSuccess: actions.billingLoaded,
```
Comment (xrdt) [inline frontend/src/scenes/billing/billingProductLogic.ts:205]: this listener never gets triggered so I moved the billingloaded call into afterMount below.

File: frontend/src/scenes/billing/BillingLimitInput.tsx [hunk 2]
```diff
@@ -20,16 +20,27 @@ export const BillingLimitInput = ({ product }: { product: BillingProductV2Type }
...
+
         if (value === undefined) {
-            return actuallyUpdateLimit()
+            // unset the limit
+            actuallyUpdateLimit()
+            return
+        }
+        if (isNaN(value) || value - Math.floor(value) !== 0) {
+            // Nan or decimal are not acceptable values.
+            setBillingLimitError('Please enter an integer.')
```
Comment (raquelmsmith) [inline frontend/src/scenes/billing/BillingLimitInput.tsx:39]: I wonder how many non-engineers will wonder what an `integer` is... it's a bit of an easter egg for the engineering type so maybe we should keep it, or maybe we should just say "please enter a whole number"

File: frontend/src/scenes/billing/BillingLimitInput.tsx [hunk 3]
```diff
@@ -20,16 +20,27 @@ export const BillingLimitInput = ({ product }: { product: BillingProductV2Type }
     )
     const { setIsEditingBillingLimit, setBillingLimitInput } = useActions(billingProductLogic({ product }))
 
+    const [billinglimitError, setBillingLimitError] = useState('')
```
Comment (raquelmsmith) [inline frontend/src/scenes/billing/BillingLimitInput.tsx:23]: this should actually probably be a value in the billingProductLogic .And this should actually be a kea form, which has built-in error handling 🙈 I'll let you decide if you want to do that refactor.

**Review by zlwaterfield — APPROVED** [review]
Looks good thank you!

**Pull request merged**